src/Foundation/NSUrlSessionHandler.cs

```diff
@@ -536,84 +536,97 @@ protected override async Task<HttpResponseMessage> SendAsync (HttpRequestMessage
 		public bool PreAuthenticate {
-			get => throw new PlatformNotSupportedException ();
-			set => throw new PlatformNotSupportedException ();
+			get => false;
+			set { }
 		}
 
-		public IDictionary<string, object> Properties {
-			get => throw new PlatformNotSupportedException ();
-		}
+		// We're ignoring this property, just like Xamarin.Android does:
+		// https://github.com/xamarin/xamarin-android/blob/09e8cb5c07ea6c39383185a3f90e53186749b802/src/Mono.Android/Xamarin.Android.Net/AndroidMessageHandler.cs#L167
+		public IDictionary<string, object> Properties { get { return null; } }
 
+		// We dont support any custom proxies, and don't let anybody wonder why their proxy isn't
+		// being used if they try to assign one (in any case we also return false from 'SupportsProxy').
 		public IWebProxy Proxy {
-			get => throw new PlatformNotSupportedException ();
+			get => null;
 			set => throw new PlatformNotSupportedException ();
 		}
 
-		public SslProtocols SslProtocols {
-			get => throw new PlatformNotSupportedException ();
-			set => throw new PlatformNotSupportedException ();
-		}
+		// There doesn't seem to be a trivial way to specify the protocols to accept (or not)
+		// It might be possible to reject some protocols in code during the challenge phase,
+		// but accepting earlier (unsafe) protocols requires adding entires to the Info.plist,
+		// which means it's not trivial to detect/accept/reject from code here.
+		// Currently the default for Apple platforms is to accept TLS v1.2 and v1.3, so default
+		// to that value, and ignore any changes to it.
+		public SslProtocols SslProtocols { get; set; } = SslProtocols.Tls12 | SslProtocols.Tls13;
 
-		public Func<HttpRequestMessage, X509Certificate2, X509Chain, SslPolicyErrors, bool> ServerCertificateCustomValidationCallback {
-			get => throw new PlatformNotSupportedException ();
-			set => throw new PlatformNotSupportedException ();
-		}
+		// We're ignoring this property, just like Xamarin.Android does:
+		// https://github.com/xamarin/xamarin-android/blob/09e8cb5c07ea6c39383185a3f90e53186749b802/src/Mono.Android/Xamarin.Android.Net/AndroidMessageHandler.cs#L160
+		public Func<HttpRequestMessage, X509Certificate2, X509Chain, SslPolicyErrors, bool> ServerCertificateCustomValidationCallback { get; set; }
 
+		// There's no way to turn off automatic decompression, so yes, we support it
 		public bool SupportsAutomaticDecompression {
-			get => throw new PlatformNotSupportedException ();
+			get => true;
 		}
 
+		// We don't support using custom proxies, but NSUrlSession will automatically use any proxies configured in the OS.
 		public bool SupportsProxy {
-			get => throw new PlatformNotSupportedException ();
+			get => false;
 		}
 
+		// We support the AllowAutoRedirect property, but we don't support changing the MaxAutomaticRedirections value.
 		public bool SupportsRedirectConfiguration {
-			get => throw new PlatformNotSupportedException ();
+			get => true;
 		}
```

Comment:
My impression was it isn't supported to _configure_ the redirection?

Reply:
My thoughts: I would use this property to guard against the use of those unsupported / throwing properties. So I'd say it is safer to return false.